src/terminal/Grid_test.cpp

```diff
@@ -846,4 +846,24 @@ TEST_CASE("Grid.reflow.tripple", "[grid]")
         // }}}
     }
 }
+
+TEST_CASE("Grid infinite", "[grid]")
+{
+    auto grid_finite = Grid<Cell>(PageSize { LineCount(2), ColumnCount(8) }, true, LineCount(0));
+    grid_finite.setLineText(LineOffset { 0 }, "ABCDEFGH"sv);
+    grid_finite.setLineText(LineOffset { 1 }, "abcdefgh"sv);
+    grid_finite.scrollUp(LineCount { 1 });
+    REQUIRE(grid_finite.lineText(LineOffset(0)) == "abcdefgh");
+    REQUIRE(grid_finite.lineText(LineOffset(-1)) == std::string(8, ' '));
+
+    auto grid_infinite = Grid<Cell>(PageSize { LineCount(2), ColumnCount(8) }, true, LineCount(-1));
```

Comment:
```suggestion
    auto grid_infinite = Grid<Cell>(PageSize { LineCount(2), ColumnCount(8) }, true, Infinite {});
```
okay, I know this code change won't work because I think `Grid`'s constructor does not accept a `MaxHistoryLineCount` yet (see comment above), but I think that would be the best way to also visually and typed express the intent here.

Reply:
> okay, I know this code change won't work because I think `Grid`'s constructor does not accept a `MaxHistoryLineCount` yet (see comment above), but I think that would be the best way to also visually and typed express the intent here.

Changed, see 'Refactor' commit